lib/cluster.js

```diff
@@ -293,28 +293,47 @@ function masterInit() {
     cluster.emit('setup', settings);
   }
 
-  var debugPortOffset = 1;
+  var debugPortOffset = 0;
+
+  // match[1]: debug argument (e.g. --debug)
+  // match[2]: host part in debug socket (e.g. [::] or 127.0.0.1)
+  // match[3]: debug port (e.g. 5858)
+  const debugArgRegex =
+    /^(--inspect|--debug|--debug-(?:brk|port))(?:=(.+:)?(\d+))?$/;
+
+  for (var i = 0; i < process.execArgv.length; i++) {
+    if (process.execArgv[i].match(debugArgRegex)) {
```

Comment:
You could use [`RegExp#test`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/test), instead of `match`, as we don't need the groups here.